File: backend/internal/constants/deadline.go
package constants

// DeadlineType 期限类型枚举。
const (
	DeadlineTypeHearing  = "hearing"
	DeadlineTypeAppeal   = "appeal"
	DeadlineTypeEvidence = "evidence"
	DeadlineTypeFiling   = "filing"
	DeadlineTypeOther    = "other"
)

// DeadlineTypeValues 全部期限类型值。
var DeadlineTypeValues = []string{DeadlineTypeHearing, DeadlineTypeAppeal, DeadlineTypeEvidence, DeadlineTypeFiling, DeadlineTypeOther}

// DeadlineStatus 期限状态枚举。
const (
	DeadlineStatusPending   = "pending"
	DeadlineStatusCompleted = "completed"
)

// DeadlineStatusValues 全部期限状态值。
var DeadlineStatusValues = []string{DeadlineStatusPending, DeadlineStatusCompleted}

// DeadlineView 期限中心视图枚举。
const (
	DeadlineViewUpcoming  = "upcoming"
	DeadlineViewOverdue   = "overdue"
	DeadlineViewCompleted = "completed"
)

// DeadlineViewValues 全部期限中心视图值。
var DeadlineViewValues = []string{DeadlineViewUpcoming, DeadlineViewOverdue, DeadlineViewCompleted}

// IsValidDeadlineType 校验期限类型。
func IsValidDeadlineType(s string) bool {
	for _, v := range DeadlineTypeValues {
		if v == s {
			return true
		}
	}
	return false
}

// IsValidDeadlineView 校验期限中心视图。
func IsValidDeadlineView(s string) bool {
	for _, v := range DeadlineViewValues {
		if v == s {
			return true
		}
	}
	return false
}

File: frontend/src/constants/deadline.ts
// 期限类型/状态/视图枚举（与后端 backend/internal/constants/deadline.go 保持一致）
export const DeadlineType = {
  HEARING: 'hearing',
  APPEAL: 'appeal',
  EVIDENCE: 'evidence',
  TRIAL: 'trial',
  EXECUTION: 'execution',
  LIMITATION: 'limitation',
  REPLY: 'reply',
  REGISTRATION: 'registration',
  OTHER: 'other',
} as const

export const DeadlineTypeText: Record<string, string> = {
  [DeadlineType.HEARING]: '开庭',
  [DeadlineType.APPEAL]: '上诉',
  [DeadlineType.EVIDENCE]: '举证',
  [DeadlineType.TRIAL]: '审理',
  [DeadlineType.EXECUTION]: '执行',
  [DeadlineType.LIMITATION]: '诉讼时效',
  [DeadlineType.REPLY]: '答辩',
  [DeadlineType.REGISTRATION]: '立案登记',
  [DeadlineType.OTHER]: '其他',
}

export const DeadlineStatus = {
  PENDING: 'pending',
  COMPLETED: 'completed',
} as const

export const DeadlineStatusText: Record<string, string> = {
  [DeadlineStatus.PENDING]: '未完成',
  [DeadlineStatus.COMPLETED]: '已完成',
}

// 期限中心视图
export const DeadlineView = {
  UPCOMING: 'upcoming',
  OVERDUE: 'overdue',
  COMPLETED: 'completed',
} as const

export const DeadlineViewText: Record<string, string> = {
  [DeadlineView.UPCOMING]: '即将到期',
  [DeadlineView.OVERDUE]: '已逾期',
  [DeadlineView.COMPLETED]: '已完成',
}

export const DeadlineTypeOptions = Object.entries(DeadlineTypeText).map(([value, label]) => ({ label, value }))
export const DeadlineViewOptions = Object.entries(DeadlineViewText).map(([value, label]) => ({ label, value }))
